handler.go

```diff
@@ -79,15 +79,37 @@ func WithOIDC(securitySchemeName, oidcURL string) Option {
 // you don't want to hardcode the hostname in your spec.
 func WithAddServerUrls() Option {
 	return func(handler *Handler) {
-		handler.middlewares = append(handler.middlewares, func(ctx *gin.Context, spec Spec) {
-			path := ctx.Request.URL.Path
-			path = strings.TrimSuffix(path, "/openapi.yml")
-			httpsUrl := fmt.Sprintf("https://%s%s", ctx.Request.Host, path)
-			httpUrl := fmt.Sprintf("http://%s%s", ctx.Request.Host, path)
-			spec.AddServerUrl(httpUrl)
-			spec.AddServerUrl(httpsUrl)
-		})
+		mw := newServerURLMiddleware(false)
+		handler.middlewares = append(handler.middlewares, mw)
+	}
+}
+
+// WithReplaceServerUrls is the same as WithAddServerUrls but it will replace all previously
+// defined server urls.
+func WithReplaceServerUrls() Option {
+	return func(handler *Handler) {
+		mw := newServerURLMiddleware(true)
+		handler.middlewares = append(handler.middlewares, mw)
+	}
+}
+
+// newServerURLMiddleware returns a new HandlerMiddleware which adds the requests host to the OpenAPI specs.
+//
```

Comment:
empty line to make it look better in IDE?

Reply:
Correct. Goland separates paragraphs only if there is an empty line in between. I think this is the same behaviour as with the official [pkg.go.dev](https://pkg.go.dev/github.com/DEXPRO-Solutions-GmbH/swaggerui#Spec.SetOpenIdConnectUrl) and Markdown where paragraphs require an empty line.